components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediatorSerializer.java

```diff
@@ -0,0 +1,161 @@
+                onCacheHit.addAttribute(
+                        fac.createOMAttribute(CachingConstants.SEQUENCE_STRING, nullNS,
+                                              cacheMediator.getOnCacheHitRef()));
+                cacheElem.addChild(onCacheHit);
+            } else if (cacheMediator.getOnCacheHitSequence() != null) {
+                onCacheHit = fac.createOMElement(CachingConstants.ON_CACHE_HIT_STRING, synNS);
+                serializeChildren(onCacheHit, cacheMediator.getOnCacheHitSequence().getList());
+                cacheElem.addChild(onCacheHit);
+            }
+
+            OMElement protocolElem = fac.createOMElement(CachingConstants.PROTOCOL_STRING, synNS);
+            protocolElem.addAttribute(
+                    fac.createOMAttribute(CachingConstants.TYPE_STRING, nullNS, cacheMediator.getProtocolType()));
+            if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(cacheMediator.getProtocolType())) {
+
+                String[] methods = cacheMediator.getHTTPMethodsToCache();
+                if (!(methods.length == 0 && "".equals(methods[0]))) {
+                    StringBuilder method = new StringBuilder();
+                    for (int i = 0; i < methods.length; i++) {
+                        if (i != methods.length - 1) {
+                            method.append(methods[i]).append(", ");
+                        } else {
+                            method.append(methods[i]);
+                        }
+                    }
+                    OMElement methodElem = fac.createOMElement(CachingConstants.METHODS_STRING, synNS);
+                    methodElem.setText(method.toString());
+                    protocolElem.addChild(methodElem);
+                }
+
+                String[] headers = cacheMediator.getHeadersToExcludeInHash();
+                if (!(headers.length == 0 && "".equals(headers[0]))) {
+                    StringBuilder header = new StringBuilder();
+                    for (int i = 0; i < headers.length; i++) {
+                        if (i != headers.length - 1) {
+                            header.append(headers[i]).append(", ");
+                        } else {
+                            header.append(headers[i]);
+                        }
+                    }
+                    OMElement headerElem = fac.createOMElement(CachingConstants.HEADERS_TO_EXCLUDE_STRING, synNS);
+                    headerElem.setText(header.toString());
+                    protocolElem.addChild(headerElem);
+                }
+
+                String responseCodes = cacheMediator.getResponseCodes();
+                OMElement responseCodesElem = fac.createOMElement(CachingConstants.RESPONSE_CODES_STRING, synNS);
+                responseCodesElem.setText(responseCodes);
+                protocolElem.addChild(responseCodesElem);
+
+            }
+
+            if (cacheMediator.getDigestGenerator() != null) {
```

Comment:
The null check is to be on the safe side. If the user hadn't included the hashGenerator it takes the default value.